andela/artemis-ah-backend
# #164198489 Signup users

#### What does this PR do?
Have user signup into the app

#### Description of Task to be completed?
Having the endpoint  below working:
* POST  `host:port/api/users`
* GET  `host:port/api/users/verifyemail`
>- User should be able to sign up
>- User should be able to receive notification upon signup
>- User should be able to verify email after receiving notification

#### How should this be manually tested?
After cloning the repo, cd into it then:
* RUN `npm install` and `npm test` to see test result
* RUN `npm run start:dev` to start the server

* Request object:
```
{
	"firstname": "John",
	"lastname": "Doe",
	"email": "john.doe@gmail.com",
	"username":"john45",
	"password": "12345678"
}
```

Using Postman test the endpoint below:
* POST  `host:port/api/users` and 
test the second route from the email you will receive, by clicking the `verify email` link
* GET  `host:port/api/users/verifyemail` with the header:
_key:_ Content-Type value: application/json

#### Any background context you want to provide?
* None
#### What are the relevant pivotal tracker stories?
[#164198489](https://www.pivotaltracker.com/story/show/164198489)

#### Screenshots (if appropriate):
* VerifyEmail:
<img width="1436" alt="screenshot 2019-03-01 at 3 27 00 am" src="https://user-images.githubusercontent.com/38951888/53612302-17cd1d00-3bd2-11e9-9773-b6d871a4819f.png">

* Response:
<img width="1440" alt="screenshot 2019-03-01 at 3 28 17 am" src="https://user-images.githubusercontent.com/38951888/53612313-2287b200-3bd2-11e9-976e-ebc714f975fa.png">
